lib/ASTGen/Sources/ASTGen/ASTGen.swift

```diff
@@ -253,6 +263,23 @@ extension ASTGenVisitor {
     return generateSourceRange(node)
   }
 
+  /// Obtains bridged token source range for a syntax node.
+  /// Unlike `generateSourceRange(_:)`, this correctly emulates the string/regex literal token SourceLoc in AST.
```

Comment:
Is there any reason to not have `generateSourceRange` handle this case?

Reply:
Performance concern, I believe `endTok.parent?.kind` still implies retain/release. Also, after fully migrating to SwiftParser/ASTGen, I believe `generateSourceRange()` is more correct. So I hesitate to use this function extensively.